### PR TITLE
Read and write of scc files in C++

### PR DESCRIPTION
**Pull request by hschreiber**

### CODE REVIEW

**Review by DavidLapous — CHANGES_REQUESTED** [review]
LGTM, apart from an issue with `strip_comments=False` (no issues if `True`)
```py
s.write_to_scc_file("truc.scc", strip_comments=False)
s = mp.Slicer()
s._build_from_scc_file("truc.scc")
```
which (up to adding an `except +` in the `slicer.pxd` file) throws with
```
Wrong format. The number of cells in each dimension is wrong.
```
I guess the reader function doesn't ignore the comments.

**Comment by hschreiber** [comment]
> LGTM, apart from an issue with `strip_comments=False` (no issues if `True`)
> 
> ```python
> s.write_to_scc_file("truc.scc", strip_comments=False)
> s = mp.Slicer()
> s._build_from_scc_file("truc.scc")
> ```
> 
> which (up to adding an `except +` in the `slicer.pxd` file) throws with
> 
> ```
> Wrong format. The number of cells in each dimension is wrong.
> ```
> 
> I guess the reader function doesn't ignore the comments.

The comments where ignored, but the format check was placed to soon. And after adding the possibility of shifting for the dimensions, the check became also wrong and I forgot to remove it.
It should work now.

File: multipers/slicer.pxd.tp [hunk 1]
```diff
@@ -85,7 +85,7 @@ cdef extern from "Persistence_slices_interface.h":
       {{endif}}
 
       {{if not IS_KCRITICAL}}
-      void build_from_scc_file(const string&, bool, bool, int) nogil
+      void build_from_scc_file(const string&, bool, bool, int) except + nogil
       void write_to_scc_file(const string&, int, int, bool, bool, bool, bool)
```
Comment (DavidLapous) [inline multipers/slicer.pxd.tp:89]: `nogil`?

**Review by DavidLapous — APPROVED** [review]

**Comment by DavidLapous** [comment]
interface should be ok now

**Comment by DavidLapous** [comment]
LGTM. I will fix test later

**Pull request merged**